pkg/config/database.go

```diff
@@ -47,5 +48,26 @@ func (d *Database) Open(logger *zap.SugaredLogger) (*icingadb.DB, error) {
 		return utils.Key(s, '_')
 	})
 
-	return icingadb.NewDb(db, logger, &d.Options), nil
+	return icingadb.NewDb(db, logger, d.Options), nil
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (d *Database) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// Prevent recursion.
+	type T Database
+	self := (*T)(d)
+
+	if err := unmarshal(&self); err != nil {
+		return internal.CantUnmarshalYAML(err, d)
+	}
+	if self.Options == nil {
+		// Options is nil if no option is set in the configuration.
+		// in order for the default values to be set.
+		// We have to call unmarshal ourselves to trigger the Options.UnmarshalYAML call
+		if err := unmarshal(&self.Options); err != nil {
```

Comment:
This shouldn't call `unmarshal()` here as this will decode the whole `database:` block into the `options:` sub-block, i.e. it now also uses this value (without the `options:` sub-key):
```yaml
database:
  max_connections: 42
```